components/suggestion/command_provider/command_provider.tsx

```diff
@@ -140,20 +173,20 @@ export default class CommandProvider extends Provider {
                                 hint = cmd.auto_complete_hint;
                             }
                             matches.push({
-                                suggestion: s,
-                                complete: '',
-                                hint,
-                                description: cmd.auto_complete_desc,
-                                iconData: '',
+                                Suggestion: s,
```

Comment:
(3/5) Why are these becoming capitalized? I'm not sure why ESLint isn't picking them up, but that violates JS naming conventions

Reply:
The existing/previous autocomplete suggestions coming from the server are in capitalized form. Notice the absence of `json` tags [here](https://github.com/mattermost/mattermost-server/blob/8248477319fb0ec3d8cf765bda2bf8341a5149e2/model/command_autocomplete.go#L28).

This change is solving an inconsistent casing and some bugs (that existed on webapp and mobile). Before this change here, the lower case version of these keys were transient among the functions on the client. Now there is one consistent model across all platforms (server, webapp, mobile) that has the capitalized version.

Reply:
@mickmister The missing json tags seem like something we should fix. Would you be open on filling a ticket for that?

Reply:
@hanzei Fixing this on server side has several implications. Apart of making the corresponding changes in webapp and mobile, we would need to put another flag in mobile to to distinguish between server versions (or make a dirty conversion of `suggestion.hint = in.hint || in.Hint;`). 0/5 we should make this fix.

Reply:
@hanzei This is an artifact from the original command autocomplete implementation, and would be a breaking change if we changed that. 3/5 we should leave it as it currently stands on all platforms. I have gone back and forth on converting them to lowercase or not on the client over the past few months, and this definitely results is the simplest code all around.

Reply:
> The missing json tags seem like something we should fix.

Adding the json tags is fine, as long as we keep the casing as uppercase in the tags to avoid breaking changes

Reply:
Oh that's ugly. I didn't know the capitalized names were an existing issue. Ideally, I think we'd stick to fixing the field names as soon as they reach the client like we did before, but I could see a pretty convincing argument to remain consistent with the field names throughout the web app. 0/5 on which way we go with it as long as we aren't spreading this pattern elsewhere.

Reply:
The pattern won't be spreading anywhere else besides code that handles slash command suggestions, which is currently scoped to the command provider and app command parser. Any other new code that may be handling this stuff in the future would be doing the same lowercase translations. The mobile app was actually using the capital letters throughout its slash command component before the apps framework changes as well.

The main reason why this works well is because all code is using the same typescript'd struct coming from the server. There's no ambiguity of possibly using some other struct that doesn't match what's coming from the server.